### ISSUE TITLE
Downloads are not shown in Download Bar nor about:downloads

### ISSUE THREAD

**Issue by btlechowski** [brave/browser-laptop#14398]
### Steps to Reproduce

  1. Go to https://github.com/brave/browser-laptop/releases
  2. Download a dmg file

**Actual result:**
The Download bar is not shown.
The download is not listed in about:downloads
Note by  @srirambv: The file is downloaded.

**Expected result:**
The Download bar is shown.
The download is shown in the Download bar
The download is listed in about:downloads

**Reproduces how often:**
always

### Brave Version
```
Brave: 0.23.8 
V8: 6.6.346.32 
rev: 06c657bb7ee5530f93c37f51cc6b7bf158912d04 
Muon: 6.1.5 
OS Release: 4.13.0-21-generic 
Update Channel: Beta 
OS Architecture: x64 
OS Platform: Linux 
Node.js: 7.9.0 
Brave Sync: v1.4.2 
libchromiumcontent: 66.0.3359.181
```

**Reproducible on current live release:**
No

cc @bsclifton @kjozwiak @LaurenWags @GeetaSarvadnya

**Comment by LaurenWags** [comment]
Reproduces for me on macOS when downloading from brave.com/download (or https://laptop-updates.brave.com/latest/osx). It looks like my file is downloaded (it is displayed in my downloads folder) however, once I close Brave-Beta (0.23.8) the file is no longer in my downloads folder. I tried downloading from another site (https://www.thinkbroadband.com/download) and the download bar was displayed and my file remained in my downloads folder after closing Brave-Beta. Note - Brave crashes on close after the download.

**Comment by srirambv** [comment]
Still an issue on the latest beta build. Download doesn't show in about:downloads or on the download bar. But the file is downloaded, closing browser window deletes the download as well.

**Comment by bsclifton** [comment]
@srirambv just to clarify: are you saying all downloads? or specifically downloads in a private tab with Tor?

**Comment by srirambv** [comment]
@bsclifton Yes. I tried downloading flash and it didn't show the download and deleted the file after the browser was closed

**Comment by kjozwiak** [comment]
@bsclifton here's an example of the issue happening on `macOS 10.13.4 x64` using the following build:

* `0.23.11` 6565c0689f8bfbd6f4d03e2e8c68c9381da157f6 
* muon: `7.1.0`  
* libchromiumcontent: `67.0.3396.87`

![downloadissue](https://user-images.githubusercontent.com/2602313/41667626-6bc504be-747b-11e8-81a1-d695f172e6ee.gif)

STR that I usually use to reproduce this pretty consistently:

* launch `0.23.11` 6565c0689f8bfbd6f4d03e2e8c68c9381da157f6 with a new profile
* visit https://github.com/brave/browser-laptop/releases
* click on one of the `dmg` files and you'll notice that there's no download bar nor logged information under `about:downloads`
* check your download location and you'll see the `dmg`
* close `0.23.11` and the download will also be removed

**Comment by petemill** [comment]
Investigation shows that the `session.on('will-download', ... )` event is not firing. This seems to only occur with muon 7.1.x and not 7.0.x

**Comment by bsclifton** [comment]
Assigning @jumde to take a look- may have been related to recent safe browsing fixes

**Comment by srirambv** [comment]
@bsclifton @darkdh Does this need the new muon ? I still don't see downloads bar or download progress on taskbar.
![14398](https://user-images.githubusercontent.com/17010094/41859039-45f14e54-7869-11e8-8116-d76f51fc6f85.gif)

**Comment by darkdh** [comment]
yes, you will need muon 7.1.2
